infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -286,17 +295,33 @@ private ServerContext getServerContext(Map<String, Object> opts) {
         // Bind to all interfaces only if no address or interface specified by the user.
         // Fetch the address if given a network interface.
         if (opts.get("--network-interface") != null) {
-            opts.put("--address", getAddressFromInterfaceName((String) opts.get("--network-interface")));
+            opts.put(
+                    ADDRESS_PARAM,
+                    getAddressFromInterfaceName(
+                            (String) opts.get("--network-interface"),
+                            (opts.get(NETWORK_INTERFACE_VERSION_PARAM) != null) ?
+                                    NetworkInterfaceVersion.valueOf(((String) opts.get(NETWORK_INTERFACE_VERSION_PARAM)).toUpperCase()):
+                                    NetworkInterfaceVersion.IPV6 // Default is IPV6
+                    )
+            );
             opts.put("--bind-to-all-interfaces", false);
-        } else if (opts.get("--address") == null) {
-            // Default the address to localhost and set the bind to all interfaces flag to true,
-            // if the address and interface is not specified.
+        } else if (opts.get(ADDRESS_PARAM) == null) {
+            // If the address and interface is not specified,
+            // pick an address from eth0 interface and set the bind to all interfaces flag to true.
             opts.put("--bind-to-all-interfaces", true);
-            opts.put("--address", "localhost");
+            opts.put(ADDRESS_PARAM,
+                    getAddressFromInterfaceName(
```

Comment:
is there a way to tell if this method will return localhost?

Reply:
This is the case when no address (ex- "127.0.0.1") or interface (ex- "eth0") is mentioned.

In this case, we start off with a viable local address from any of the generic interfaces (eth0, en0, eth1, en1,..., lo). Whichever address we pick, it doesn't really matter as the previous line, 
`opts.put("--bind-to-all-interfaces", true);` 
makes CorfuServer listen on all addresses on port 9000 (from args). Clients will be able to connect to all addresses + port 9000 and bootstrap the server with a specific layout.